src/client/datascience/notebook/kernelProvider.ts

```diff
@@ -94,6 +98,7 @@ export class VSCodeKernelPickerProvider implements INotebookKernelProvider {
         webview: NotebookCommunication,
         token: CancellationToken
     ): Promise<void> {
+        webview.onDidReceiveMessage((m) => this._onDidGetFontAwesomeMessage.fire({ webview, message: m }));
```

Comment:
I think we need to dispose this (or use the 3rd parameter here that is a list of disposables)

Reply:
```suggestion
webview.onDidReceiveMessage((m) => this._onDidReceiveMessage.fire({ webview, message: m }), undefined, this.disposables);
```

Reply:
I would prefer we moved the code from `kernelWithMetadata` into this method here directly. This way we do'nt need to add a new event & an event handler, less changes as well & wee know exactly what's going on.

Reply:
I suggested we added an event because the handler for the message is the thing that knows about preloads. Otherwise the idea of fontAwesome propagates out to another class.